Src/IronPython.Modules/_codecs.cs

```diff
@@ -122,7 +122,7 @@ public static PythonTuple ascii_encode(CodeContext context, string input, string
 
         public static object decode(CodeContext/*!*/ context, object obj, string encoding = null, string errors = "strict") {
             if (encoding == null) {
-                encoding = context.LanguageContext.DefaultEncoding.EncodingName;
+                encoding = context.LanguageContext.GetDefaultEncodingName();
```

Comment:
I haven't looked into this, but any idea why we wouldn't just use the default encoding directly with `DoDecode` instead of going via `lookup`?

Reply:
I suppose you are right. If I had to guess, I would say it was because with `setdefaultencoding` it was possible to set any encoding as default, including a user-defined one. 

I haven't looked into this either, except noticing that there are a number of things in `_codecs` that don't seem quite right. I was thinking that once I start pulling on this thread, more things start to unravel, which are better left for a separate PR. So my intention for this WIP was to have minimal changes to `_codecs` that will just let this PR to pass the tests.

I will implement your suggestion now if the tests still pass.